Gems/Terrain/Code/Source/Components/TerrainWorldDebuggerComponent.cpp

```diff
@@ -395,5 +424,27 @@ namespace Terrain
         }
     }
 
+    void TerrainWorldDebuggerComponent::WireframeSector::Reset()
+    {
+        AZStd::lock_guard<AZStd::recursive_mutex> lock(m_sectorStateMutex);
+        if (m_jobContext)
+        {
+            m_jobContext->Cancel();
+            m_jobContext.reset();
```

Comment:
Cancel does not block; I think it should be fine because the callback is handling the case where the sector has become dirty again by just bailing out, but if you're worried about it I can add something to make it block here until the completion callback gets invoked.

Reply:
My concern is really with debugger destruction, not refreshes. As long as we have a way to block when destroying the debugger, we should be ok, but otherwise I think there's a chance for race conditions here.